File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        // if(target > nums[n - 1]) return n;
        // if(target < nums[0]) return 0;
        int l = 0, r = nums.length - 1;
        while(l <= r) {
            int m = (r - l) / 2 + l;
            if(nums[m] == target) return m;
            else if(nums[m] > target) r = m - 1;
            else l = m + 1;
        }
        return l;
    }
}

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        left, right = 0, len(nums) - 1
        while left <= right:
            m = (left + right) // 2
            if nums[m] == target:
                return m
            if target < nums[m]:
                right = m - 1
            else:
                left = m + 1
        return left
        